app.go

```diff
@@ -62,26 +62,30 @@ type App interface {
 	Lifecycle() Lifecycle
 }
 
-var app App
-var appLock sync.RWMutex
+// app is a App typed variable. Note that we cannot use atomic.Value as
```

Comment:
I think I understand this comment but it is a little confusing.
Maybe "app contains an App variable, but due to atomic.Value restrictions on interfaces we need to use an indirect type `internalApp`"

Reply:
Should be resolved in the latest commit.